spec/facter/resolvers/networking_linux_spec.rb

```diff
@@ -458,6 +465,13 @@ def ifaddr_obj(name, ip, mac, netmask, ipv4_type)
         Facter::Util::Resolvers::Networking::PrimaryInterface.instance_variable_set(:@log, log_spy)
         allow(Facter::Util::FileHelper).to receive(:safe_read).with('/proc/net/route', '').and_return('')
         allow(Facter::Core::Execution).to receive(:which).with('ip').and_return('ip available')
+        allow(Facter::Core::Execution).to receive(:execute)
+          .with('ip -json -6 route show default', logger: log_spy)
+          .and_return(load_fixture('ip_-json_-6_route_show_default').read)
+      end
+
+      it 'returns primary6 interface' do
+        expect(networking_linux.resolve(:primary6_interface)).to eq('ens3')
```

Comment:
This unit test fails because after the call to `Facter::Util::FileHelper` with `'/proc/net/route', ''` is made, the next one will be `Facter::Core::Execution.execute('ip route show default', logger: @log)` from `PrimaryInterface.read_from_ip_route`. So you need another stub for this call.